File: app/src/main/java/com/cornell/daily/sun/PostObject.java
package com.cornell.daily.sun;

import java.util.*;
//import com.cornell.daily.sun.AuthorObject;
//import com.cornell.daily.sun.FeaturedMediaImages;
//import com.cornell.daily.sun.PostAttachmentObject;

public class PostObject {

    //Bookmark data
    Date storeDate;
    boolean didSave;

    //Post data
    int id;
    Date date;
    String title;
    String content;
    String excerpt;
    String link;
    AuthorObject[] author;
    FeaturedMediaImages featuredMediaImages;
    String featuredMediaCaption;
    String featuredMediaCredit;
    String[] categories;
    String primaryCategory;
    String[] tags;
    PostAttachmentObject[] postAttachments;
    PostType postType;
    SuggestedStoryObject[] suggestedStories;
}

File: app/src/main/java/com/cornell/daily/sun/data/PostObject.kt
package com.cornell.daily.sun.data

import java.util.*

data class PostObject(
    val storeDate: Date?,
    val didSave: Boolean?,
    val id: Int,
    val title: String?,
    val content: String?,
    val excerpt: String?,
    val link: String?,
    val authors: List<AuthorObject>?,
    val featuredMediaImages: FeaturedMediaImages?,
    val featuredMediaCaption: String?,
    val featuredMediaCredit: String?,
    val categories: List<String>?,
    val primaryCategory: String?,
    val tags: List<String>,
    val postAttachments: List<PostAttachmentObject>,
    val postType: PostType?,
    val suggestedStories: List<SuggestedStoryObject>
)
